compiler/qsc_frontend/src/compile.rs

```diff
@@ -530,3 +572,46 @@ fn assert_no_errors(sources: &SourceMap, errors: &mut Vec<Error>) {
         panic!("could not compile package");
     }
 }
+
+#[must_use]
+pub fn longest_common_prefix_old<'a>(strs: &'a [&'a str]) -> &'a str {
+    let Some(common_prefix_so_far) = strs.first() else {
+        return "";
+    };
+
+    for (i, character) in common_prefix_so_far.chars().enumerate() {
+        for string in strs {
+            if string.chars().nth(i) != Some(character) {
+                return &common_prefix_so_far[0..i];
+            }
+        }
+    }
+
+    common_prefix_so_far
+}
+
+#[must_use]
+pub fn longest_common_prefix<'a>(strs: &'a [&'a str]) -> &'a str {
```

Comment:
If there's only 1 name in the list, without a directory, I'd expect this to return `""`, not the whole string. Otherwise `relative_project_sources()` returns empty string which is pretty counterintuitive.

It's also more consistent with the behavior when there are 2 or more files.

e.g. 

* `["a", "b"]` => `""`
* `["a"]` => `""`
* `["a/b", "a/c"]` => `"a/"`
* `["a/b"]` => `"a/"`

Reply:
In test form, I'd expect all these to pass, but none of them do. The 4th one panics.

```rs
#[test]
fn test_longest_common_prefix_two_relative_paths() {
    expect!["a/"].assert_eq(longest_common_prefix(&["a/b", "a/c"]));
}

#[test]
fn test_longest_common_prefix_one_relative_path() {
    expect!["a/"].assert_eq(longest_common_prefix(&["a/b"]));
}

#[test]
fn test_longest_common_prefix_one_file_name() {
    expect![""].assert_eq(longest_common_prefix(&["a"]));
}

#[test]
fn test_longest_common_prefix_only_root_common() {
    expect![""].assert_eq(longest_common_prefix(&["/a/b", "/b/c"]));
}
```